### PR TITLE
19720: added cron for creating lessons

### PR DESCRIPTION
**Pull request by nechtobolshee**

### CODE REVIEW

File: english/cron.py [hunk 1]
```diff
@@ -14,3 +17,11 @@ class ChangeLessonsStatusJob:
...
         time_end__lte=datetime.now()
     ).update(_status=Lessons.DONE)
+
+
+class CreateLessonsJob:
+    for item in Class.objects.all():
+        try:
+            item.create_lessons()
+        except Exception as e:
+            pass
```
Comment (YevheniiaSmyrnova) [inline english/cron.py:27]: Add error log

File: english/cron.py [hunk 2]
```diff
@@ -14,3 +17,11 @@ class ChangeLessonsStatusJob:
     Lessons.objects.exclude(_status__in=[Lessons.CANCELED, Lessons.DONE]).filter(
         time_end__lte=datetime.now()
     ).update(_status=Lessons.DONE)
+
+
+class CreateLessonsJob:
+    for item in Class.objects.all():
```
Comment (YevheniiaSmyrnova) [inline english/cron.py:23]: item is not good name

File: english/cron.py [hunk 3]
```diff
@@ -14,3 +17,11 @@ class ChangeLessonsStatusJob:
...
+        try:
+            item.create_lessons()
+        except Exception as e:
```
Comment (YevheniiaSmyrnova) [inline english/cron.py:26]: e is not very good name

File: english/models.py [hunk 4]
```diff
@@ -21,6 +23,28 @@ class Meta:
...
+        scheduled_days = selected_schedule.days
+        for day in range(1, (date_end - date_from).days + 1):
+            date = date_from + timedelta(days=day)
+            if date.weekday() in scheduled_days:
+                lessons_to_create.append(Lessons(
+                    class_name=self,
+                    time_start=datetime.combine(date=date, time=selected_schedule.start_time),
+                    time_end=datetime.combine(date=date, time=selected_schedule.end_time),
+                ))
+        if len(lessons_to_create) > 0:
```
Comment (YevheniiaSmyrnova) [inline english/models.py:44]: if len == 0 you do not need use bulk_create

File: english/models.py [hunk 5]
```diff
@@ -31,27 +55,34 @@ def formfield(self, **kwargs):
         defaults.update(kwargs)
         return super(ArrayField, self).formfield(**defaults)
 
+    def to_python(self, value):
```
Comment (YevheniiaSmyrnova) [inline english/models.py:58]: Why you need it?

File: english/models.py [hunk 6]
```diff
@@ -31,27 +55,34 @@ def formfield(self, **kwargs):
...
+    class_name = models.OneToOneField(Class, on_delete=models.CASCADE, verbose_name="Class name", related_name="schedule")
+    days = ChoiceArrayField(base_field=models.IntegerField(choices=week_days), default=list, size=5, blank=True)
+    start_time = models.TimeField(auto_now=False, auto_now_add=False)
+    end_time = models.TimeField(auto_now=False, auto_now_add=False)
 
     class Meta:
         verbose_name_plural = "Schedule"
 
-    def __str__(self):
+    def str(self):
```
Comment (YevheniiaSmyrnova) [inline english/models.py:85]: Why you change it?

File: english/models.py (hunk 5 above)
Comment (nechtobolshee) [reply english/models.py:58]: To store integers in postresql arrayfield. 
Without that it raises an error

File: english/models.py (hunk 6 above)
Comment (nechtobolshee) [reply english/models.py:85]: merge issue. fixed

**Review by YevheniiaSmyrnova — APPROVED** [review]

**Pull request merged**